### ISSUE TITLE
Cannot use as an authorizer lambda on AWS

### ISSUE THREAD

**Issue by Twometer** [spring-cloud/spring-cloud-function#822]
**Describe the bug**
I'm on Spring Cloud 2021.0.0, and I want to build a Spring Cloud Function to use as an authorizer lambda on AWS.

These types of lambdas have the same input type as a API Gateway Proxy, so I can use the `APIGatewayProxyRequestEvent` as the input type. However, it requires an `IamPolicyResponse` as the response type. The Spring Cloud Function AWS runtime always wraps my output in an `APIGatewayProxyResponseEvent` which makes my lambda fail.

Is there a way to turn this feaure off? Or is this a bug that it falsely detects I am in a proxy? Or am I missing something else?

**Sample**
```kotlin
 @Bean
fun authorizeRequest(): Function<APIGatewayProxyRequestEvent, IamPolicyResponse> = Function { event ->
    // ...
    IamPolicyResponse.builder()
            .withPrincipalId(accountId)
            .withPolicyDocument(policy)
            .withContext(mapOf("test" to 123))
            .build()
}
```

will always return the `IamPolicyResponse` object wrapped in another object. Never happened with lambdas with a different input type.

**Comment by Twometer** [comment]
Is there not a way to deactivate this automatic API Gateway detection?

**Comment by Twometer** [comment]
@olegz I think #827 should fix it

**Comment by Twometer** [comment]
@olegz thank you! To also fix it in the 3.2.x version, should I make a separate pull request?
